tests/cross/src/test/scala/tests/pc/CompletionArgSuite.scala

```diff
@@ -244,6 +244,111 @@ class CompletionArgSuite extends BaseCompletionSuite {
     "",
   )
 
+  check(
+    "default-args",
+    """|object Main {
+       |  def foo() = {
+       |    def deployment(
+       |      fst: String,
+       |      snd: Int = 1,
+       |    ): Option[Int] = ???
+       |    val abc = deployment(@@)
+       |  }
+       |}
+       |""".stripMargin,
+    """|fst = : String
+       |snd = : Int
+       |""".stripMargin,
+    topLines = Some(2),
+  )
+
+  check(
+    "default-args2",
+    """|object Main {
+       |  def deployment(
+       |    fst: String,
+       |    snd: Int = 1,
+       |  ): Option[Int] = ???
+       |  val abc = deployment(@@)
+       |}
+       |""".stripMargin,
+    """|fst = : String
+       |snd = : Int
+       |""".stripMargin,
+    topLines = Some(2),
+  )
+
+  check(
+    "default-args3",
+    """|object Main {
+       |  def deployment(str: String)(
+       |    fst: String,
+       |    snd: Int = 1,
+       |  ): Option[Int] = ???
+       |  val abc = deployment("str")(
+       |    @@
+       |  )
+       |}
+       |""".stripMargin,
+    """|fst = : String
+       |snd = : Int
+       |""".stripMargin,
+    topLines = Some(2),
+  )
+
+  check(
+    "default-args4",
+    """|object Main {
+       |  def deployment(str: String)(opt: Option[Int])(
+       |    fst: String,
+       |    snd: Int = 1,
+       |  ): Option[Int] = ???
+       |  val abc = deployment("str")(None)(
+       |    @@
+       |  )
+       |}
+       |""".stripMargin,
+    """|fst = : String
+       |snd = : Int
+       |""".stripMargin,
+    topLines = Some(2),
+  )
+
+  check(
+    "default-args5",
```

Comment:
Before `3.3.1`, the path in this case looks way different - 
After `3.3.1`:
```
  Ident(name = CURSOR),
  Apply(
    fun = Apply(fun = Ident(name = deployment), args = List(Literal(const = ( = "str")))),
    args = List(Ident(name = CURSOR))
  ),
  ValDef(
    name = abc,
    tpt = TypeTree[dotty.tools.dotc.core.Types$UnspecifiedErrorType$@2de07d9c],
    preRhs = Apply(
      fun = Apply(fun = Ident(name = deployment), args = List(Literal(const = ( = "str")))),
      args = List(Ident(name = CURSOR))
    )
  )
```
Before: 
```
  Ident(name = CURSOR),
  ValDef(
    name = DerivedName(underlying = opt, info = NumberedInfo(num = 1)),
    tpt = TypeTree[dotty.tools.dotc.core.Types$PreviousErrorType@44aa396b],
    preRhs = Ident(name = CURSOR)
  ),
  Block(
    stats = List(
      ValDef(
        name = DerivedName(underlying = opt, info = NumberedInfo(num = 1)),
        tpt = TypeTree[dotty.tools.dotc.core.Types$PreviousErrorType@44aa396b],
        preRhs = Ident(name = CURSOR)
      )
    ),
    expr = Block(
      stats = List(
        DefDef(
          name = $anonfun,
          paramss = List(
            List(
              ValDef(
                name = fst,
                tpt = TypeTree[TypeRef(TermRef(ThisType(TypeRef(NoPrefix,module class scala)),object Predef),type String)],
                preRhs = Thicket(trees = List())
              ),
              ValDef(
                name = snd,
                tpt = TypeTree[TypeRef(TermRef(ThisType(TypeRef(NoPrefix,module class <root>)),object scala),class Int)],
                preRhs = Thicket(trees = List())
              )
            )
          ),
          tpt = TypeTree[AppliedType(TypeRef(TermRef(ThisType(TypeRef(NoPrefix,module class <root>)),object scala),class Option),List(TypeRef(TermRef(ThisType(TypeRef(NoPrefix,module class <root>)),object scala),class Int)))],
          preRhs = Apply(
            fun = Apply(
              fun = Apply(fun = Ident(name = deployment), args = List(Literal(const = ( = "str")))),
              args = List(Ident(name = DerivedName(underlying = opt, info = NumberedInfo(num = 1))))
            ),
            args = List(Ident(name = fst), Ident(name = snd))
          )
        )
      ),
      expr = Closure(env = List(), meth = Ident(name = $anonfun), tpt = Thicket(trees = List()))
    )
  )
```
I will investigate further what can be done for older version